shared/fs/send-link-to-chat/container.js

```diff
@@ -1 +1,138 @@
 // @flow
+import * as I from 'immutable'
+import {namedConnect} from '../../util/container'
+import * as Types from '../../constants/types/fs'
+import * as ChatTypes from '../../constants/types/chat2'
+import * as ChatConstants from '../../constants/chat2'
+import * as TeamsTypes from '../../constants/types/teams'
+import * as TeamsConstants from '../../constants/teams'
+import * as Constants from '../../constants/fs'
+import * as FsGen from '../../actions/fs-gen'
+import * as ChatGen from '../../actions/chat2-gen'
+import HiddenString from '../../util/hidden-string'
+import {navigateUp, putActionIfOnPath} from '../../actions/route-tree'
+import SendLinkToChat from '.'
+
+type OwnProps = {
+  routePath: I.List<string>,
+}
+
+const _getChannelInfosOrEmptyMap = (state): I.Map<ChatTypes.ConversationIDKey, TeamsTypes.ChannelInfo> => {
+  const elems = Types.getPathElements(state.fs.sendLinkToChat.path)
+  if (elems.length < 3 || elems[1] !== 'team') {
+    return I.Map()
+  }
+  return TeamsConstants.getTeamChannelInfos(state, elems[2])
+}
+
+const mapStateToProps = state => ({
+  _channelInfos: _getChannelInfosOrEmptyMap(state),
+  _username: state.config.username,
+  sendLinkToChat: state.fs.sendLinkToChat,
+})
+
+const mapDispatchToProps = (dispatch, ownProps) => ({
+  _selectChannel: (convID: ChatTypes.ConversationIDKey) =>
+    dispatch(FsGen.createSetSendLinkToChatConvID({convID})),
+  _send: (conversationIDKey: ChatTypes.ConversationIDKey, text: string) => {
+    dispatch(ChatGen.createMessageSend({conversationIDKey, text: new HiddenString(text)}))
+    dispatch(putActionIfOnPath(ownProps.routePath, navigateUp()))
```

Comment:
why navigate up when you're just navigating to the thread after

Reply:
This way if user clicks the Files tab from the Chat tab, they'd see the folder view. Otherwise the SendLinkToChat modal would still be shown. Not showing it anymore in this case made more sense to me, but please let me know if you think otherwise.